packages/dappmanager/src/modules/upnpc/upnpError.ts

```diff
@@ -1,45 +1,43 @@
-// ERROR - no UPNP DEVICE:
+export function parseUpnpErrors(terminalOutput: string): UpnpError {
+  if (isUpnpAvailable(terminalOutput))
+    return new UpnpError(terminalOutput, UPnPErrors.UPNPNOTAVAIL);
+  if (isNotExistingPort(terminalOutput))
+    return new UpnpError(terminalOutput, UPnPErrors.UPNPNOTPORT);
+  return new UpnpError(terminalOutput, UPnPErrors.UNKNOWN);
+}
+
+function isUpnpAvailable(terminalOutput: string): boolean {
+  return !(
+    RegExp(/No IGD UPnP Device found/, "i").test(terminalOutput) ||
+    RegExp(/No valid UPNP Internet Gateway Device/, "i").test(terminalOutput)
+  );
+}
 
-// upnpc : miniupnpc library test client, version 2.0.
-//  (c) 2005-2017 Thomas Bernard.
-// Go to http://miniupnp.free.fr/ or https://miniupnp.tuxfamily.org/
-// for more information.
-// No IGD UPnP Device found on the network !
+function isNotExistingPort(terminalOutput: string): boolean {
+  const lines = terminalOutput.trim().split(/\r?\n/);
+  const lastLine = lines[lines.length - 1] || "";
+
+  // Check if it contains "failed"
+  return lastLine.includes("failed");
+}
 
 export class UpnpError extends Error {
-  command: string;
   terminalOutput: string;
-  isUpnpAvailable: boolean;
-  isPortExisting: boolean;
+  typeUpnpError: UPnPErrors;
 
-  constructor({
-    terminalOutput,
-    command
-  }: {
-    terminalOutput: string;
-    command: string;
-  }) {
-    super(`UPnP error. Command failed: ${command}. Error: ${terminalOutput}`);
-    this.command = command;
+  constructor(terminalOutput: string, typeUpnpError: UPnPErrors) {
+    super(`UPnP Error: ${typeUpnpError}\nError:${terminalOutput}`);
     this.terminalOutput = terminalOutput;
-    this.isUpnpAvailable = this.getUpnpAvailable();
-    this.isPortExisting = this.getPortExists();
+    this.typeUpnpError = typeUpnpError;
   }
+}
 
-  getUpnpAvailable(): boolean {
-    return !(
-      RegExp(/No IGD UPnP Device found/, "i").test(this.terminalOutput) ||
-      RegExp(/No valid UPNP Internet Gateway Device/, "i").test(
-        this.terminalOutput
-      )
-    );
-  }
-
-  getPortExists(): boolean {
-    const lines = this.terminalOutput.trim().split(/\r?\n/);
-    const lastLine = lines[lines.length - 1] || "";
-
-    // Check if it contains "failed"
-    return lastLine.includes("failed");
-  }
+// UPNPNOTAVAIL: UPnP is not available
+// UPNPNOTPORT: UPnP not existing port
+// UNKNOWN: UPnP unclassified error
+// Research for more errors!
+export enum UPnPErrors {
+  "UPNPNOTAVAIL",
```

Comment:
No need to use NodeJS garbage acronyms haha. You can use `UPNP_NOT_AVAILABLE`, space is free!